### PR TITLE
Web questions

### PR DESCRIPTION
**Pull request by Toxicc05**

### CODE REVIEW

File: docs/module_web_with_python.md [hunk 1]
```diff
@@ -5,137 +5,437 @@
 ### Clean code
 
 #### Point out 5 suggestions, how to format an SQL query!
+* Separate your query into multiple lines.
+* Capitalize keywords such as "WHERE, HAVING, SELECT" for better readability.
+* Use "AS" to rename columns when needed for a prettier display.
+* Use underscore when naming tables/columns with multiple words in them.
+* Avoid giving the same name for both the table and a column.
 #### What layers can you name in a simple web application?
```
Comment (WailGree) [inline docs/module_web_with_python.md:13]: Might worth to note other form of layers:
Presentation,
Data service,
Business,
Data access

File: docs/module_web_with_python.md [hunk 2]
```diff
@@ -5,137 +5,437 @@
...
+* Front-end (What happens on the user's computer.)
+* Back-end (What happens on the server's side.)
+* Data storage and access (Usually SQL Datbase, or CSV file.)
 ### Error handling
 #### What error can occur, when an array does not have an element on the requested index?
+* If the requested index is bigger than the length of the array, then an index error occurs.
+* Else, the program runs as it would normally, however, the returned value will be empty.
 #### What is the “finally” block, and how would you use it?
+* The finally block is part of a try-except block.
+* A try-except block is used to catch special exceptions types in a program.
```
Comment (WailGree) [inline docs/module_web_with_python.md:23]: Typo "exceptions" -> "exception"

File: docs/module_web_with_python.md [hunk 3]
```diff
@@ -5,137 +5,437 @@
...
-
+* Front-end (What happens on the user's computer.)
+* Back-end (What happens on the server's side.)
+* Data storage and access (Usually SQL Datbase, or CSV file.)
 ### Error handling
 #### What error can occur, when an array does not have an element on the requested index?
+* If the requested index is bigger than the length of the array, then an index error occurs.
+* Else, the program runs as it would normally, however, the returned value will be empty.
 #### What is the “finally” block, and how would you use it?
+* The finally block is part of a try-except block.
```
Comment (WailGree) [inline docs/module_web_with_python.md:22]: Might worth to also mention, as try-catch, as it's more often heard of in programming.

File: docs/module_web_with_python.md [hunk 4]
```diff
@@ -5,137 +5,437 @@
...
+* SQL injection is a code injection technique in which a client inserts malicious SQL statements into an entry field for execution.
+* To protect your applications from SQL Injection. Check all data coming from the user and use scripts or libraries to catch keywords in the data.
+* example: Hello, my name is'); DROP DATABASE users;
 #### What is XSS? How to protect an application against it?
+* XSS, Cross-Site-Scripting is a security breach which takes advantage of dynamically generated webpages.
+* An XSS attack happens when a client injects malicious third party scripts into a dynamically generated website.
+* Most common script types include: Javascript, HTML, Flash <- now obsolete.
+* Via XSS, the client is able to: highjack accounts, change user settings, and/or poison cookies with malicious code.
 #### How to properly store passwords?
+* Password should be stored after put through a hashing (encoding) algorithm for increased security.
```
Comment (WailGree) [inline docs/module_web_with_python.md:39]: Might worth to note that salting them after hashing is also recommended.

File: docs/module_web_with_python.md [hunk 5]
```diff
@@ -5,137 +5,437 @@
...
 #### What is HTTPS?
+* Secure, encripted Hyper Text Transport Protocol connection handler.
```
Comment (WailGree) [inline docs/module_web_with_python.md:41]: Typo "encripted" -> "encrypted"

File: docs/module_web_with_python.md [hunk 6]
```diff
@@ -5,137 +5,437 @@
...
+* Stacks follow a first in, last out principle, meaning the last element inserted will be returned first.
+* Elements can only be inserted at the first place of the stack. This element is then tracked as the top.
+* Queues follow a first in first out principle, meaning the first element inserted is the first to get returned.
+* Elements can only be added at the rear (end) of the queue.
 #### What is BubbleSort? Describe the main logic of this sorting algorithm.
+* Bubble sorting works by repeatedly swapping the adjesent elements of a list if they are in the wrong order.
+* The sorting algorithm has to go through the list without changing any element to know that it's sorted.
 #### Explain the process of finding the maximum and minimum value in a list of numbers!
+* We declare two variables, a minimum and a maximum and make it equal to the first element in the list.
+* Then we iterate over the list and when we find a number that is lower than our 'minimum' variable or bigger than our 'maximum' variable, then we swap the minimum's or the maxumum's value accordingly.
```
Comment (WailGree) [inline docs/module_web_with_python.md:79]: Typo "maxumum" -> "maximum"

File: docs/module_web_with_python.md [hunk 7]
```diff
@@ -5,137 +5,437 @@
...
+	x.addEventListener('click', function(event){ body of the function })
 #### How to trigger an event in JavaScript?
+* The first argument in the event listener hints to the method of how to trigger it's function.
+* 'click' <-- function triggers upon clicking at the target,
+* 'dragstart' <-- function triggers upon starting to drag the object,
+* 'dragover' <-- function triggers upon dragging something over the event target.
+* 'dragend' <-- function triggers upon deciding to end the dragging of the event target,
+* 'drop' <-- function triggers upon dropping the event target to somewhere other than where it started from originally.
 #### What is a callback function? Tell some examples of its usage.
+* Callback are when a function is used as argument in another function.
```
Comment (WailGree) [inline docs/module_web_with_python.md:155]: Typo:
Version a: "Callback" -> "Callbacks"
Version b: "Callback are" -> "Callback is"

File: docs/module_web_with_python.md [hunk 8]
```diff
@@ -5,137 +5,437 @@
...
+* import defaultExport from "module-name"; <-- imports the defaultExport from a module.
+* import { export1 } from "module-name"; <--Import a single export from a module.
+* import { export1 as alias1 } from "module-name"; <-- Import a single export from a module and rename it.
+* import { export1 , export2 } from "module-name"; <-- Import multiple exports from a single module.
 
 ### Functional
 #### What is recursion?
+* A recursive  function is when a function calls itself again based on a set of conditions.
 #### Write a recursive function which calculates the Fibonacci numbers!
+*	def fib(n):
```
Comment (WailGree) [inline docs/module_web_with_python.md:130]: Forgot to format code for Python. Code is written in one line -> harder to read.

File: docs/module_web_with_python.md [hunk 9]
```diff
@@ -5,137 +5,437 @@
...
+* SQL aggreagte functions operate on a set of values and returns a single value.
+* AVG() - returns the average of a set.
+* COUNT() - returns the number of items in a set.
+* SUM() - returns the sum of all or distinct values in a set.
 #### What kind of JOIN types do you know in SQL? Could you give examples?
+* INNER JOIN: Returns records that have matching values in both tables
+* LEFT JOIN: Returns all records from the left table, and the matched records from the right table
+* RIGHT JOIN: Returns all records from the right table, and the matched records from the left table
+* FULL JOIN: Returns all records when there is a match in either left or right table
 #### What are the constraints in sql?
```
Comment (WailGree) [inline docs/module_web_with_python.md:203]: Typo "sql" -> "SQL"

File: docs/module_web_with_python.md [hunk 10]
```diff
@@ -5,137 +5,437 @@
...
 #### What are database indexes? When to use?
+* A database index is a data structure that improves the speed of data retrieval operations on a database at the cost of additional writes and storage space to maintain the index data structure.
+* An index normally includes a 'key' or direct link to the original row of data from which it was copied.
+* We index databases when we want to improve the speed of data operations inside the database.
 #### What are database transactions? When to use?
+* A database transaction symbolizes a unit of work performed within a database management system against a database.
+* To provide isolation between programs accessing a database concurrently.
+* For recovery when an execution stops completely or partially and some or many operations remain incomplete.
 #### What kind of database relations do you know? How to define them?
+* One-to-One: A row in table A can have only one matching row in table B, and vice versa.
```
Comment (WailGree) [inline docs/module_web_with_python.md:226]: Worth to mention "many-to-many" relations.

File: docs/module_web_with_python.md [hunk 11]
```diff
@@ -5,137 +5,437 @@
...
+* One-to-many: This is the most common relationship type. In this type of relationship, a row in table A can have many matching rows in table B, but a row in table B can have only one matching row in table A.
+(Each customer can only be assigned to one city. One city can be assigned to many customers.)*
 #### You have a table with an “address” field which contains data like “3525, Miskolc, Régiposta 9.” (postcode, city, street name and address). How would you query all records related to Miskolc?
+* SELECT * from address WHERE city LIKE '%iskolc' or postcode = 3525.
```
Comment (WailGree) [inline docs/module_web_with_python.md:230]: SQL query not meeting question.

File: docs/module_web_with_python.md [hunk 12]
```diff
@@ -5,137 +5,437 @@
...
 ### Javascript
 
 #### What is javascript?
+* JavaScript is a high-level, object oriented programming language.
+* Javascript is used to control the program behavior of webpages.
+* Javascript is often compiled on-the-run when a client loads a webpage.
 #### When to use AJAX? Bring examples of its usage.
+* We use AJAX when we want to send or receive data from the server to the client without reloading the page.
 #### What is DOM and how to manipulate it from Javascript?
+* DOM stands for Document Oject Model for html and XML.
```
Comment (WailGree) [inline docs/module_web_with_python.md:290]: Typo "html" -> "HTML"

File: docs/module_web_with_python.md [hunk 13]
```diff
@@ -5,137 +5,437 @@
...
+* examples for events: "click", "dragover", "dragstart", "dragend"
 #### What is event bubbling/capturing? How would you use it?
+* Event bubbling happens when an inner scope function is called while an outer scope DOM element also has an event listener tied to it on the same event trigger.
+* Event bubbling means that all functions will activate one by one starting from the innermost scope.
+* Capturing is done to prevent the events from triggering all at once.
+* example: event.cancelBubble = true; or event.stopPropagation();
 #### What is JSON and how do we use it?
-
+* JSON stands for JavaScript Object Notation.
+* JSON is an open standard  file format and data interchange format.
```
Comment (WailGree) [inline docs/module_web_with_python.md:306]: Typo "standard  file" -> "standard file" (double spaces)

File: docs/module_web_with_python.md [hunk 14]
```diff
@@ -5,137 +5,437 @@
...
 #### What is your favorite git command? Why?
+* Commit, because when I'm using it, it means that I've progressed with my tasks.
 #### What does remote/local mean in Git? 
+* Local means files and changes that are only on your computer and are not shared.
+* Remote are the files and changes that are shared between every collaborator in the project.
 
 ### DevOps
 
 #### Why is it good to use a package manager software?
+* Installing softwares are made way easier by their existence.
```
Comment (WailGree) [inline docs/module_web_with_python.md:333]: Typo "softwares are" -> "software is". Optional.

File: docs/module_web_with_python.md [hunk 15]
```diff
@@ -5,137 +5,437 @@
...
+* With package managers, you are able to roll back to previous versions of programs with ease.
 #### Why is it good to use a virtual environment for a project?
+* Runtime enviroment updates can break programs when done due to differing syntaxes.
+* With virtual enviroments you can make sure that the projects enviroment stays the same as in the beggining, even if you upgrade your main runtime enviroment.
```
Comment (WailGree) [inline docs/module_web_with_python.md:337]: Typo "beggining" -> "beginning"

File: docs/module_web_with_python.md [hunk 16]
```diff
@@ -5,137 +5,437 @@
...
+	-HTTP Status Code (For example HTTP/1.1 301 Moved Permanently, means the requested resource was permanently moved and redirecting 		to some other resource).
+	-Headers (Example – Content-Type: html)
+	-An empty line.
+	-A message body which is optional.
 #### What is DNS? How does it work?
+* DNS stands for Domain Name Service.
+* DNS services act like phonebooks for the internet.
+* When you search for 'youtube.com' in your browser, your request gets sent through a DNS server, that DNS server then translates your url and sends you to the corresponding IP address.
 #### What is a web server?
+* A web server is a computer and software combination that is connected to the internet and accepts HTTP requests do distribute web pages.
```
Comment (WailGree) [inline docs/module_web_with_python.md:383]: Typo "do" -> "to"

File: docs/module_web_with_python.md [hunk 17]
```diff
@@ -5,137 +5,437 @@
...
 #### What would you use a session for?
+* To monitor and manage clients' permissions on the server.
+* To cherrypick data that is specific to that exact user.
 #### What would you use a cookie for?
-
+* To store information about a client such as: login credentials, permissions, and on-site search history.
 ## Software Development Methodologies
 
 #### What kind of software development methodologies do you know? What are the main features of these?
+* It’s a rigid linear model that consists of sequential phases (requirements, design, implementation, verification, maintenance) in which distinct goals are accomplished.
```
Comment (WailGree) [inline docs/module_web_with_python.md:397]: Missing naming for Waterfall method (?)

**Review by WailGree — APPROVED** [review]
Found smaller typos, and a few places where addition would be advised. Also please check the SQL query. No real reason to say "Not passed" though. :)

File: docs/module_web_with_python.md (hunk 17 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:397]: FIxed

File: docs/module_web_with_python.md (hunk 16 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:383]: fixed

File: docs/module_web_with_python.md (hunk 15 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:337]: fixed

File: docs/module_web_with_python.md (hunk 14 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:333]: fixed

File: docs/module_web_with_python.md (hunk 13 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:306]: fixed

File: docs/module_web_with_python.md (hunk 1 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:13]: fixed

File: docs/module_web_with_python.md (hunk 2 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:23]: fixed

File: docs/module_web_with_python.md (hunk 3 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:22]: fixed

File: docs/module_web_with_python.md (hunk 4 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:39]: fixed

File: docs/module_web_with_python.md (hunk 8 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:130]: ?

File: docs/module_web_with_python.md (hunk 9 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:203]: fixed

File: docs/module_web_with_python.md (hunk 11 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:230]: fixed, hopefully

File: docs/module_web_with_python.md (hunk 12 above)
Comment (Toxicc05) [reply docs/module_web_with_python.md:290]: fixed

**Pull request merged**